api/lib/validate.js

```diff
@@ -0,0 +1,59 @@
+const _ = require('lodash');
+const errors = require('./errors.js');
+const validator = require('./schema-validator');
+const ripple = require('ripple-lib');
+
+function addOptionalOption(validator) {
+  return function(parameter, optional) {
+    if (optional && parameter === undefined) {
+      return null;
+    }
+    return validator(parameter);
+  };
+}
+
+function validateAccount(account) {
+  if (!ripple.UInt160.is_valid(account)) {
+    return new errors.InvalidRequestError(
```

Comment:
Yes, it is part of `api`. We should rename the error classes, but I think that can be a separate pull request.